bravado/mapping/param.py

```diff
@@ -128,6 +132,73 @@ def marshal_param(param, value, request):
             format(location))
 
 
+def unmarshal_param(param, request):
+    """Unmarshal the given parameter from the passed in request like object.
+
+    :type param: :class:`bravado.mapping.param.Param`
+    :type request: :class:`bravado.mapping.request.RequestLike`
+    """
+    param_spec = get_param_type_spec(param)
+    location = param.location
+    cast_param = partial(cast_request_param, param_spec['type'], param.name)
+
+    if location == 'path':
+        raw_value = cast_param(request.path.get(param.name, None))
```

Comment:
May be i am missing something, why do we need to cast each param from `request_object`? When creating a request `client.getPetById(petId=42)` `petId` should already be in its correct form? When does it convert to `string`?